File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head == null || head.next == null){
            return head;
        }
        ListNode first = head, second = head;
        while(second != null && second.next != null){
            second = second.next;
            int temp = first.val;
            first.val = second.val;
            second.val = temp;
            first = first.next.next;
            second = second.next;
        }
        return head;
    }
}

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0)
        dummy.next = head

        prev = dummy
        first = head

        while first and first.next:
            second = first.next
            _next = second.next
            #swap
            prev.next = second
            second.next = first
            first.next = _next
            prev = first
            first = _next

        return dummy.next